apiserver/root.go

```diff
@@ -828,9 +840,10 @@ func (ctx *facadeContext) ModelImporter() facade.ModelImporter {
 	pool := ctx.r.shared.statePool
 	return migration.NewModelImporter(
 		state.NewController(pool),
-		ctx.migrationScope(),
+		ctx.migrationScope,
+		ctx.ServiceFactory().ModelManager(),
 		ctx.ServiceFactory().ControllerConfig(),
-		ctx.ServiceFactory().Machine(),
+		ctx.ServiceFactoryGetter(),
```

Comment:
Is there any way to not expose this as a public method? I've nicely hidden this behind a public API `ModelExporter` and `ModelImporter` on the `facade.Context` interface.

I believe you could even do `ctx.r.serviceFactoryGetter` and you get the win win.

Reply:
done